ThunderCloud/InlineButtonView.swift

```diff
@@ -12,48 +12,110 @@ import ThunderTable
-
-	/// The `TSCLink` to determine what action is performed when the button is pressed
-	open var link: StormLink?
-	
-	/// A Bool to disable and enable the button
-	open var isAvailable: Bool = false {
-		didSet {
-			style()
-		}
-	}
-	
-	public override init(frame: CGRect) {
-		
-		super.init(frame: frame)
-		cornerRadius = 8.0
-		titleLabel?.textAlignment = .center
-		style()
-	}
-	
-	required public init?(coder aDecoder: NSCoder) {
-		super.init(coder: aDecoder)
-		style()
-	}
-	
-	override open var intrinsicContentSize: CGSize {
-		let superSize = super.intrinsicContentSize
-		return CGSize(width: superSize.width, height: max(superSize.height, 44))
-	}
-	
-	open func style() {
-		
-		borderWidth = 1.0
-		let mainColor = ThemeManager.shared.theme.mainColor
-		
-		if !isAvailable {
-			primaryColor = mainColor.withAlphaComponent(0.2)
-			secondaryColor = mainColor.withAlphaComponent(0.2)
-			isUserInteractionEnabled = false
-		} else {
-			primaryColor = mainColor
-			secondaryColor = mainColor
-			isUserInteractionEnabled = true
-		}
-	}
+    
+    /// The `TSCLink` to determine what action is performed when the button is pressed
+    open var link: StormLink?
+    
+    open var progressView: UIImageView?
+    
+    /// A Bool to disable and enable the button
+    open var isAvailable: Bool = false {
+        didSet {
+            style()
+        }
+    }
+    
+    public override init(frame: CGRect) {
+        
+        super.init(frame: frame)
+        cornerRadius = 8.0
+        titleLabel?.textAlignment = .center
+        style()
+    }
+    
+    required public init?(coder aDecoder: NSCoder) {
+        super.init(coder: aDecoder)
+        style()
+    }
+    
+    override open var intrinsicContentSize: CGSize {
+        let superSize = super.intrinsicContentSize
+        return CGSize(width: superSize.width, height: max(superSize.height, 44))
+    }
+    
+    open func style() {
+        
+        borderWidth = 1.0
+        let mainColor = ThemeManager.shared.theme.mainColor
+        
+        if !isAvailable {
+            primaryColor = mainColor.withAlphaComponent(0.2)
+            secondaryColor = mainColor.withAlphaComponent(0.2)
+            isUserInteractionEnabled = false
+        } else {
+            primaryColor = mainColor
+            secondaryColor = mainColor
+            isUserInteractionEnabled = true
+        }
+    }
+    
+    open func stopTimer() {
+        
+        if let borderColor = layer.borderColor {
+            setTitleColor(UIColor(cgColor: borderColor), for: .normal)
+        }
+        
+        UIView.transition(with: self, duration: 0.15, options: .transitionCrossDissolve, animations: { [weak self] in
+            self?.progressView?.removeFromSuperview()
+            self?.setTitle("Start Timer".localised(with: "_STORM_TIMER_START_TITLE"), for: .normal)
+            }, completion: nil)
+    }
+    
+    open func startTimer() {
+        
+        let bundle = Bundle(for: InlineButtonView.self)
```

Comment:
If this has been subclassed, will it still use the ThunderCloud bundle?

Reply:
It will, I think this is acceptable behaviour. The only place we've *ever* overriden it is BRC FA and it doesn't use the same progress view!